SEP-0012.md

```diff
@@ -0,0 +1,343 @@
+        """
+        Returns the WCS physical types associated with each array axis.
+
+        Returns an iterable of tuples where each tuple corresponds to an array axis and
+        holds strings denoting the WCS physical types associated with that array axis.
+        Since multiple physical types can be associated with one array axis, tuples can
+        be of different lengths. Likewise, as a single physical type can correspond to
+        multiple array axes, the same physical type string can appear in multiple tuples.
+        """
+
+    @abc.abstractmethod
+    def axis_world_coords(self,
+                          *axes: Union[Union[int, str], Iterable[Union[int, str]]],
+                          edges: bool = False) -> Iterable[object]:
+        """
+        Returns world coordinate values of all pixels for all axes.
+
+        Parameters
+        ----------
+        axes
+            Axis number in numpy ordering or unique substring of
+            `.NDCube.wcs.world_axis_physical_types` or
+            `.NDCube.wcs.world_axis_names` of axes for which real world
+            coordinates are desired. ``axes=None`` implies all axes will be
+            returned.
+
+        edges
+            The edges argument helps in returning ``pixel_edges``
+            instead of ``pixel_values``. Default value is False,
+            which returns ``pixel_values``. True return ``pixel_edges``
+
+        Returns
+        -------
+        axes_coords
+            High level object, or iterable thereof, giving the real world
+            coords for the axes requested by user.
+            For example, SkyCoords. The types returned are determined by the WCS object.
+
+        Example
+        -------
+        >>> NDCube.all_world_coords(('lat', 'lon'))
+        >>> NDCube.all_world_coords(2)
+
+        """
+
+    @abc.abstractmethod
+    def axis_world_coords_values(self,
+                                 *axes: Union[Union[int, str], Iterable[Union[int, str]]],
+                                 edges: bool = False) -> Iterable[u.Quantity]:
+        """
+        Returns world coordinate values of all pixels for all axes.
+
+        Parameters
+        ----------
+        axes
+            Axis number in numpy ordering or unique substring of
+            `.NDCube.wcs.world_axis_physical_types` or
+            `.NDCube.wcs.world_axis_names` of axes for which real world
+            coordinates are desired. ``axes=None`` implies all axes will be
+            returned.
+
+        edges
+            The edges argument helps in returning ``pixel_edges``
+            instead of ``pixel_values``. Default value is False,
+            which returns ``pixel_values``. True return ``pixel_edges``
+
+        Returns
+        -------
+        axes_coords
+            `~astropy.units.Quantity` or iterable thereof for all requested
+            world axes, units determined by the wcs.  # TODO: Should we mandate namedtuple?
```

Comment:
* Same question as above.  Should an iterable always be returned?

* I vote we DO mandate a ```namedtuple``` because of the significant possibility of ambiguity in determining which returned Quantity corresponds to which axis/physical type.  Even for experienced ndcube developers, this will probably serve as a very useful sanity check at times.

Reply:
mandating the namedtuple in the ABC adds a significant amount of complexity to the ABC specification. Given that for `axes=False` its `zip(return self.array_axis_physical_types)` I am not sure it's worth the burden.

Reply:
@DanRyanIrish and I discussed this out of band and agreed that namedtuple while good isn't a core part of the API.